### ISSUE TITLE
1.28.1 - Crashes when opening a package's settings in the settings view

### ISSUE THREAD

**Issue by calebmeyer** [atom/atom#17663]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Atom crashes whenever I open (in safe mode or not) package settings. Not every time, but about every 2 or 3 packages

### Steps to Reproduce

1. Open atom
2. `Ctrl-,` to open the settings view
3. Click Packages on the left
4. Wait for the packages to load
5. Click into the settings for a package
6. Click packages again
7. Repeat 5 and 6 until crash

**Expected behavior:** Atom is stable

**Actual behavior:** Atom crashes

**Reproduces how often:** 
I've been able to reproduce this frequently, maybe every other or third package I click into. Atom crashes on launch after as well.

### Versions
![image](https://user-images.githubusercontent.com/3342777/42517893-6bce2fde-8426-11e8-8b7b-849b3907b9e7.png)

My full settings: https://gist.github.com/calebmeyer/57593e1d220fef2910e3d881852a04b3

### Additional Information

Here's an mp4 of the crash: [crash.zip](https://github.com/atom/atom/files/2177892/crash.zip)

Linked from #17595 per @lee-dohm's request

**Comment by calebmeyer** [comment]
I'm wondering if there's something up with my config.cson, so I'm doing the .atom -> .atom-backup thing.

**Comment by calebmeyer** [comment]
I did the reset, re-installed my packages with apm, and immediately reproduced the crash.

**Comment by lee-dohm** [comment]
@jasonrudolph Just wanted to put this on your radar.

**Comment by jasonrudolph** [comment]
:wave: @calebmeyer: Thanks for letting us know about this issue. I'm wondering if this is the same issue that we're tracking in https://github.com/atom/atom/issues/17557. Can you tell us if you're using a proxy?

**Comment by jasonrudolph** [comment]
Cross-referencing https://github.com/Microsoft/vscode/issues/48022, in which VS Code is experiencing a similar issue when viewing "extensions" (i.e., the VS Code term for "packages") following their upgrade to Electron 2.0

**Comment by calebmeyer** [comment]
@jasonrudolph I've run the editor at work, on my work laptop. So our network may be behind some sort of proxy. I haven't had our dedicated VPN running, but windows has this "corporate connection" thing running. 

I'll see if I can repro on my desktop at home tonight.

**Comment by calebmeyer** [comment]
I went through a bunch of package settings (1.28.1) on my home desktop, and I was unable to reproduce this issue. I looked through the thread on 17557, and tried unchecking the apm proxy setting, it did not help.

I then went to windows 10's proxy settings. Apparently I am behind a proxy and didn't know it. Turning it off fixes this issue, and I haven't been able to detect anything breaking yet.

![image](https://user-images.githubusercontent.com/3342777/42577254-9c585fee-84e9-11e8-98a9-87957915b3f2.png)

Thanks for your all's help debugging this!
